lib/ansible/module_utils/vmware.py

```diff
@@ -234,7 +234,7 @@ def compile_folder_path_for_object(vobj):
     thisobj = vobj
     while hasattr(thisobj, 'parent'):
         thisobj = thisobj.parent
-        if thisobj.name == 'Datacenters':
+        if (not hasattr(thisobj, 'parent')) or (hasattr(thisobj, 'parent') and thisobj.parent is None):
```

Comment:
How about 

```Python
    while True:
        try:
            thisobj = thisobj.parent
        except (IndexError, AttributeError) as err:
            break
        if isinstance(thisobj, vim.Folder):
            paths.append(thisobj.name)
    paths.reverse()
    return '/' + '/'.join(paths)
```

Reply:
This works in both Python 2 and 3. Here are few [tests](https://github.com/Akasurde/pyvmomi_examples/blob/master/compile_path_example.py#L54)

Reply:
Looks pretty good to me. Will give it a try

Reply:
Cool. Let me know results for this.